cmd/signimage/signimage.go

```diff
@@ -3,21 +3,141 @@ package main
 import (
 	"archive/tar"
 	"bytes"
+	"encoding/json"
 	"flag"
 	"fmt"
-	"github.com/docker/cli/cli/config"
-	dockertypes "github.com/docker/cli/cli/config/types"
 	"github.com/go-logr/logr"
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
 	"io"
+	"io/fs"
 	"k8s.io/klog/v2/klogr"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
 )
 
+// configDir should contain all the secrets available as individual files named for their keys
+// so we need to search through for the secrets we need for our pull and push repos.
+// and if we find nothingi default anonymous auth
```

Comment:
```suggestion
// If we do not find any authn, default to anonymous.
```

Reply:
vim needs a "random 'i' detected, have you forgotten you're already in insert mode?"  warning :)
Changed.